### PR TITLE
2.7 improvement

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**

### CODE REVIEW

File: src/cmsteps.cc [hunk 1]
```diff
@@ -156,48 +186,37 @@ int cmsteps_put(cmsteps_t* self, const char* step)
...
+    return n ? int64_t(*n) : -1;
 }
 
 //  --------------------------------------------------------------------------
-//  Return iterator to first item
+//  Return iterator to first item or NULL
 
 uint32_t* cmsteps_first(cmsteps_t* self)
 {
+    assert(self);
```
Comment (ericclappier-eaton) [inline src/cmsteps.cc]: Prefer "return nullptr" as others instead of assert

File: src/cmsteps.cc [hunk 2]
```diff
@@ -156,48 +186,37 @@ int cmsteps_put(cmsteps_t* self, const char* step)
...
     return reinterpret_cast<uint32_t*>(zhashx_first(self->steps));
 }
 
 //  --------------------------------------------------------------------------
-//  Return iterator to next item or nullptr
+//  Return iterator to next item or NULL
 
 uint32_t* cmsteps_next(cmsteps_t* self)
 {
+    assert(self);
```
Comment (ericclappier-eaton) [inline src/cmsteps.cc]: Prefer "return nullptr" as others instead of assert

File: src/cmsteps.cc [hunk 3]
```diff
@@ -156,48 +186,37 @@ int cmsteps_put(cmsteps_t* self, const char* step)
...
     return reinterpret_cast<uint32_t*>(zhashx_next(self->steps));
 }
 
 //  --------------------------------------------------------------------------
-//  Return cursor
+//  Return cursor on current item
 
 const void* cmsteps_cursor(cmsteps_t* self)
 {
+    assert(self);
```
Comment (ericclappier-eaton) [inline src/cmsteps.cc]: Prefer "return nullptr" as others instead of assert

**Review by ericclappier-eaton — DISMISSED** [review]

File: src/cmsteps.cc (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply src/cmsteps.cc]: totally agree... assert() is a pain... and shall be removed definitively...
But, globally,  the caller's rarely check pointers.. and returned pointers from functions...
(here is a very special case).

Comment (FrancoisRegisDegott-eaton) [reply src/cmsteps.cc]: (all done)

**Review by ericclappier-eaton — APPROVED** [review]

**Pull request merged**